tasks/main.yml

```diff
@@ -30,6 +54,55 @@
     state: directory
   notify: Restart uchiwa
 
+- block:
+  - name: Create directory for publickey
+    file:
+      path: "{{ uchiwa_publickey_path | dirname }}"
+      owner: "{{ uchiwa_user }}"
+      group: "{{ uchiwa_group }}"
+      state: directory
+      mode: 0755
+
+  - name: Create publickey
+    copy:
+      dest: "{{ uchiwa_publickey_path }}"
+      mode: "{{ uchiwa_publickey_mode }}"
+      owner: "{{ uchiwa_user }}"
+      group: "{{ uchiwa_group }}"
+      content: "{{ uchiwa_publickey }}"
+    notify: Restart uchiwa
+  when:
+    - "'uchiwa' in uchiwa_config"
+    - "'auth' in uchiwa_config.uchiwa"
+    - "'publickey' in uchiwa_config.uchiwa.auth"
+    - uchiwa_config.uchiwa.auth.publickey | length > 0
+    - uchiwa_publickey is defined
+    - uchiwa_publickey | length > 0
+
+- block:
+  - name: Create directory for privatekey
+    file:
+      path: "{{ uchiwa_privatekey_path | dirname }}"
+      owner: "{{ uchiwa_user }}"
+      group: "{{ uchiwa_group }}"
+      state: directory
+      mode: 0755
+  - name: Create privatekey
+    copy:
+      dest: "{{ uchiwa_privatekey_path }}"
+      mode: "{{ uchiwa_privatekey_mode }}"
+      owner: "{{ uchiwa_user }}"
+      group: "{{ uchiwa_group }}"
+      content: "{{ uchiwa_privatekey }}"
+    notify: Restart uchiwa
+  when:
+    - "'uchiwa' in uchiwa_config"
```

Comment:
these four conditions are always met because of the assert task above and not necessary.